### PR TITLE
[feature]:issue15 - add try-catch for run migrations

### PR DESCRIPTION
**Pull request by EvgenyBarinov**
Fix for [issue15](https://github.com/spiral/migrations/issues/15)

### CODE REVIEW

**Comment by wolfy-j** [comment]
Any tests for it?

**Comment by EvgenyBarinov** [comment]
Tests were modified. 
Error patterns were used from [PR12](https://github.com/spiral/migrations/pull/12) thanks for @aywan .

**Review by wolfy-j — APPROVED** [review]

File: src/Migrator.php [hunk 1]
```diff
@@ -131,21 +134,36 @@ public function run(CapsuleInterface $capsule = null): ?MigrationInterface
...
+                    \sprintf(
+                        'Error in the migration (%s) occurred: %s',
+                        \sprintf(
+                            '%s (%s)',
+                            $migration->getState()->getName(),
+                            $migration->getState()->getTimeCreated()->format(self::DB_DATE_FORMAT)
+                        ),
+                        $exception->getMessage()
+                    ),
+                    $exception->getCode()
```
Comment (alexndr-novikov) [inline src/Migrator.php]: you've forgot third argument, $previous. It's a good practice to set it to previous exception when you re-throw it in catch block

**Review by alexndr-novikov — CHANGES_REQUESTED** [review]

File: src/Migrator.php (hunk 1 above)
Comment (EvgenyBarinov) [reply src/Migrator.php]: added

**Review by alexndr-novikov — APPROVED** [review]

**Pull request merged**